DEVELOPERS_GUIDE.md

```diff
@@ -0,0 +1,87 @@
+
+There is a third test that has, unfortunately, not bee applied to all parsers as of yet. In this test, the parsers will parse a log fragment to produce a well known event. The test passes if all of the attributes of that event match the expected value.
+
+
+For historical reasons, SurvivorSpace is treated as a JVMEvent even though in reality, it’s an attribute of a young generational collection. This record is created by a parser that is specialized to parse the log and extract out the survivor information.
+
+### JVMEvent
+
+A JVMEvent represents something that happened in the JVM. The primary attributes are the type of event, the time of the event, and it’s duration. Each specialization of JVMEvent will contain attributes that are specific to that event. For example, a PSYoung represent a Parallel Young generational collection. In addition to the information above, it will have heap occupancy before and after the collection along with the size of heap before and after the collection. The event may contain more information depending upon the level of detailed information in the log file. As previously mentioned, any survivor space data will be ignored as that is dealt with in its own event.  A complete set of events can be seen in the diagram below. A complete discription of each event can be found in the corresponding JavaDoc.
+
+![GCToolkit Design](images/GCToolkit_Events.png)
+Figure 2. GCToolkit Event Hierarchy
+
+### Aggregator and Aggregation
+
+
+As can be seen in Figure 1, the end point for delivery of a JVMEvent as far as the framework is concerned, is an Aggregator. Aggregator classes must be provided by the users of GCToolkit as they are not provided as they are considered out of scope for this project. The role of the Aggregator is to capture the JVMEvent objects that are of interest to it and the extract the data of interest and pass it along to an Aggregation.
+
+As is the case with Aggregator, Aggregation classes must be supplied by users of GCToolkit. The role of an Aggregation is to collect the data and perform all calculations needed to transform the event data into a form that fits the needs of the client. For example, a GUI may want to display heap occupancy after collection. The Aggregator would capture all GC pause events from which contain reasonable accurate measures of this metrics. This information would be collected into an Aggregator.
+
+    It is the responsibility of GCToolkit to load and register all instances of Aggregation via the module’s SPI
```

Comment:
Need to mention the Aggregates and Collates annotations.

I would say that Aggregation classes need to be registered with GCToolkit, and that this can be done via GCToolkit API or through SPI. The benefit of SPI is that it allows the user to add additional Aggregations without having to modify code.